mobile/src/main/java/org/openhab/habdroid/ui/ConnectionWebViewClient.kt

```diff
@@ -62,6 +71,36 @@ open class ConnectionWebViewClient(
         }
     }
 
+    override fun onReceivedClientCertRequest(view: WebView, request: ClientCertRequest) {
+        Log.d(TAG, "SSL Client Cert required")
+        val prefs = view.context.getPrefs()
+        if (prefs.isDemoModeEnabled()) {
+            request.cancel()
+            return
+        }
+
+        val alias = prefs.getString(Constants.PREFERENCE_SSL_CLIENT_CERT, null)
+        Log.d(TAG, "Using alias $alias")
+        if (alias == null) {
+            request.cancel()
+            return
+        }
+
+        GlobalScope.launch(Dispatchers.IO) {
+            try {
+                val chain = KeyChain.getCertificateChain(view.context, alias)
+                val privateKey = KeyChain.getPrivateKey(view.context, alias)
+                request.proceed(privateKey, chain)
```

Comment:
I wonder whether it's OK to do the callback on a different thread, or whether we should do it in the main thread.

Reply:
FWIW I haven't had any issues using it so far so it doesn't seem to cause any problems.
I'm actually wondering the opposite : the whole class is only used for networking stuff, shouldn't all that be happening on the IO thread ?
(Note that I don't know the first thing about kotlin, and I haven't worked on Android for ages, so I don't really know what I'm talking about.)

Reply:
I looked into the docs for this, and the current behavior seems fine, so NVM.